TravelPoints-UI/src/app/components/admin-dashboard/admin-dashboard.component.ts

```diff
@@ -69,6 +73,19 @@ export class AdminDashboardComponent implements OnInit, AfterViewInit {
     this.touristAttractionService.getAllTouristAttractions().subscribe((touristAttractions: TouristAttraction[]) => {
       this.touristAttractions = touristAttractions
     })
+    this.reviewService.getNumberOfVisitsPerAttraction().subscribe((touristAttractionVisits: TouristAttractionVisits[])=> {
+      let i = 0
+      this.touristAttractionVisits = []
+      this.touristAttractionNames = []
+      for(let touristAttractionVisit of touristAttractionVisits) {
+        this.touristAttractionVisits[i] = touristAttractionVisit.noOfVisits!
+        this.touristAttractionNames[i] = touristAttractionVisit.name!
+        i++
```

Comment:
here, you should have used forEach instead of for(let... and you could have declared the arrays as Array and used the push method instead of using an index